src/sql/workbench/services/notebook/sql/sqlSessionManager.ts

```diff
@@ -66,11 +72,15 @@ export class SqlSessionManager implements nb.SessionManager {
 	}
 
 	startNew(options: nb.ISessionOptions): Thenable<nb.ISession> {
-		let session = new SqlSession(options, this._instantiationService);
-		return Promise.resolve(session);
+		this._session = new SqlSession(options, this._instantiationService);
```

Comment:
Do you want to do anything with the old session if there is one? Seems like you should probably be disconnecting it or something